pkg/syncer/admissionhandler/admissionhandler.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admissionhandler
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
+)
+
+type (
+	parameterSet map[string]struct{}
+)
+
+// Has checks if specified paramName present in the paramSet
+func (paramSet parameterSet) Has(paramName string) bool {
+	_, ok := paramSet[paramName]
+	return ok
+}
+
+var (
+	webhookServerPort = 8443
+	server            *http.Server
+)
+
+// StartWebhookServer starts the webhook server
+func StartWebhookServer(ctx context.Context, cert string, key string, port string) error {
+	log := logger.GetLogger(ctx)
+	certs, err := tls.X509KeyPair([]byte(cert), []byte(key))
+	if err != nil {
+		log.Errorf("failed to load key pair: %v", err)
+		return err
+	}
+	if port == "" {
+		port = fmt.Sprintf("%v", webhookServerPort)
+	}
+	server = &http.Server{
+		Addr:      fmt.Sprintf(":%v", port),
+		TLSConfig: &tls.Config{Certificates: []tls.Certificate{certs}},
```

Comment:
Can we add this TLSConfig only if certs are non-empty. For SV, we don't need to run webhook server securely as CNS-CSI controller is running on master which is not accessible to any user.

Reply:
This is not true Balu. SV also starts a webhook server using a certificate in the scheduler extension.
In SV  ValidatingWebhookConfiguration uses local address `127.0.0.1`, and thus does not require to supply CABundle in the `ValidatingWebhookConfiguration`.

```
clientConfig:
    url: 'https://127.0.0.1:12345/scheduler/admission'
```